### DISCUSSION TITLE
`utoipa::openapi::OpenApi` and `utoipa::openapi::OpenApi` have similar names, but are actually distinct types

### DISCUSSION THREAD

**Discussion by SamTV12345** [juhaku/utoipa#772] [Q&A] [answered]
Hi,

I am trying to upgrade to the latest utoipa version 4 and utoipa swagger ui version 3.1.6. Unfortunately I get the below error. Does someone know why I am getting this? I thought choosing the latest version on both dependencies should work. 

```
error[E0308]: mismatched types
   --> src\main.rs:276:84
    |
276 |         .service(SwaggerUi::new("/swagger-ui/{_:.*}").url("/api-doc/openapi.json", openapi))
    |                                                       ---                          ^^^^^^^ expected `utoipa::openapi::OpenApi`, found a different `utoipa::openapi::OpenApi`
    |                                                       |
    |                                                       arguments to this method are incorrect
    |
    = note: `utoipa::openapi::OpenApi` and `utoipa::openapi::OpenApi` have similar names, but are actually distinct types
note: `utoipa::openapi::OpenApi` is defined in crate `utoipa`
   --> C:\Users\user\.cargo\registry\src\index.crates.io-6f17d22bba15001f\utoipa-4.0.0\src\openapi.rs:38:1
    |
38  | / builder! {
39  | |     /// # Examples
40  | |     ///
41  | |     /// Create [`OpenApi`] using [`OpenApiBuilder`].
...   |
116 | |     }
117 | | }
    | |_^
note: `utoipa::openapi::OpenApi` is defined in crate `utoipa`
   --> C:\Users\user\.cargo\registry\src\index.crates.io-6f17d22bba15001f\utoipa-3.5.0\src\openapi.rs:38:1
    |
38  | / builder! {
39  | |     /// # Examples
40  | |     ///
41  | |     /// Create [`OpenApi`] using [`OpenApiBuilder`].
...   |
116 | |     }
117 | | }
    | |_^
     ```

**Comment by juhaku** [comment]
The error you are facing is most likely caused by some weird cache issues. If yo run `cargo clean` command or remove manually the crates indexes (`...index.crates.io-6f17d22bba15001f\utoipa-....`) and try again it should start working.

Though at the moment there is another issue that will surface next. That is I forgot to update the latest versions of utoipa for `utoipa-swagger-ui` and `utoipa-redoc` and `utoipa-rapidoc`. I made new releases of those crates and things should be sailing smoothly by now.

Related to #469

**Reply by SamTV12345** [reply, marked as answer]
I did a cargo clean before. But version 4.0.0 for utoipa-swagger-ui  did fix the issue. Thanks for the help and the awesome work with this crate 😄 .
